epsonscan2: 6.7.63.0 -> 6.7.70.0

Shawn8901 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Updates epsonscan2 to the latest version. As flathub is not shipping that version is switched patch source from flathub to AUR.
Aside that i updated our build.patch and added a gcc14 patch (as i noticed that epsonscan2 does not build on staging-next).
In addition the icon is now showing correctly when built with `withGUI = true;` 

For testing a scanned a page with a ET-3750 using epsonscan2 GUI.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

james-atkins committed:
epsonscan2: move to pkgs/by-name

james-atkins committed:
epsonscan2: 6.7.63.0 -> 6.7.70.0

Epson doesn't seem to publish a changelog anywhere.
Remove 32-bit version as it is 2024.

Shawn8901 commented on pkgs/misc/drivers/epsonscan2/default.nix:
```diff
@@ -55,26 +55,29 @@ stdenv.mkDerivation {
   ];
 
   postPatch = ''
+    rm CMakeCache.txt
```
fyi that is needed as the currents version archive contains that cmake cache file, thus its unhappy about our build system.

Shawn8901 commented on pkgs/misc/drivers/epsonscan2/default.nix:
```diff
@@ -127,8 +130,8 @@ stdenv.mkDerivation {
       mkdir -p $out/share/icons/hicolor/{48x48,128x128}/apps
-      convert $src/Resources/Icons/escan2_app.ico -resize 48x48 $out/share/icons/hicolor/48x48/apps/epsonscan2.png
-      convert $src/Resources/Icons/escan2_app.ico -resize 128x128 $out/share/icons/hicolor/128x128/apps/epsonscan2.png
+      convert $src/Resources/Icons/escan2_app.ico -resize 48x48 -delete 1,2,3 $out/share/icons/hicolor/48x48/apps/epsonscan2.png
+      convert $src/Resources/Icons/escan2_app.ico -resize 128x128 -delete 1,2,3 $out/share/icons/hicolor/128x128/apps/epsonscan2.png
```
icons havent worked, as the ico contains multiple resolutions and resulting in multiple pngs after convertion and thus not being named correctly

Shawn8901 commented:
> Thanks! I pushed PR #358303 a while ago but it hasn't been merged and so this would supersede it. Nice catch with the icons.

ah, i did not see that pr, i was mainly motivated by the compile error with gcc14. Should i split that out so that we can cherry-pick those to your pr?

FliegendeWurst commented on pkgs/misc/drivers/epsonscan2/default.nix:
```diff
@@ -55,26 +55,29 @@ stdenv.mkDerivation {
     (fetchpatch {
-      url = "https://github.com/flathub/net.epson.epsonscan2/raw/master/patches/epsonscan2-crash.patch";
-      hash = "sha256-srMxlFfnZuJ3ed5veFcJIiZuW27F/3xOS0yr4ywn4FI=";
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/0002-Fix-crash.patch?h=epsonscan2";
```
```suggestion
      url = "https://aur.archlinux.org/cgit/aur.git/plain/0002-Fix-crash.patch?h=epsonscan2&&id=6e422c487f5f3b135df554fde65ab57bc6377159";
```
Same for other URLs, to ensure the content doesn't change.

Shawn8901 replied:
i did so, but by keeping the flatpak patch sources, as they also apply correctly on 6.7.70.0 (which i did not even check before).

Shawn8901 committed:
epsonscan2: use pinned patches, add gcc14 patch

Shawn8901 committed:
epsonscan2: fix not showing icons when built with withGui = true

upstream ico file contains multiple resolutions, those need to be deleted
on convert

Shawn8901 commented:
as staging-next was merged. That becomes an unbreak PR for master (already targets master).
